projects/packages/videopress/src/client/admin/hooks/use-search-params/index.ts

```diff
@@ -1,16 +1,73 @@
 /**
  * External dependencies
  */
-import { useLocation } from 'react-router-dom';
+import { useLocation, useHistory } from 'react-router-dom';
 
-/**
- * Gets a given parameter from the search query.
- *
- * @param {string} parameterName - The name of the parameter to get from the query string.
- * @param {string} defaultValue - The default value to return if the given parameter is not set on the query string.
- * @returns {string} - The value of the parameter if it's set. The defaultValue if the parameter is not set.
- */
-export const useSearchParam = ( parameterName: string, defaultValue: string = null ) => {
-	const searchParams = new URLSearchParams( useLocation().search );
-	return searchParams.has( parameterName ) ? searchParams.get( parameterName ) : defaultValue;
+export const useSearchParams = () => {
+	const location = useLocation();
+	const history = useHistory();
+	const searchParams = new URLSearchParams( location.search );
+
+	/**
+	 * Gets a given parameter from the search query.
+	 *
+	 * @param {string} parameterName - The name of the parameter to get from the query string.
+	 * @param {string} defaultValue - The default value to return if the given parameter is not set on the query string.
+	 * @returns {string} - The value of the parameter if it's set. The defaultValue if the parameter is not set.
+	 */
+	const getParam = ( parameterName: string, defaultValue: string = null ) => {
+		return searchParams.has( parameterName ) ? searchParams.get( parameterName ) : defaultValue;
+	};
+
+	/**
+	 * Sets a given parameter on the search query data, but does not refresh the URL.
+	 *
+	 * @param {string} parameterName - The name of the parameter to set on the query string.
+	 * @param {string} value - The value to be set for the parameter on the query string.
+	 */
+	const setParam = ( parameterName: string, value: string = null ) => {
+		searchParams.set( parameterName, value );
+	};
+
+	/**
+	 * Deletes a given parameter from the search query data, which results on removing
+	 * it from the URL when it gets updated.
+	 *
+	 * @param {string} parameterName - The name of the parameter to delete.
+	 */
+	const deleteParam = ( parameterName: string ) => {
+		searchParams.delete( parameterName );
+	};
+
+	/**
+	 * Update the URL query string with the current values of the searchParams object.
+	 */
+	const update = () => {
+		const searchFragment = '?' + searchParams.toString();
```

Comment:
Probably better to use `addQueryParams` instead to avoid setting `?` manually.

Reply:
Seems like this `addQueryParams` is a Postman thing. 

I looked for some elegant ways to not use the `?` here, and I actually could ignore it when setting the new `search` value, but I need it on the `if` statement because the original `history.location.search` also contains a trailing `?`, so I had to decide between removing the `?` from `history.location.search` to do the comparison OR add it to `searchFragment`...